solidity/contracts/token/ReserveToken.sol

```diff
@@ -0,0 +1,133 @@
+     */
+    function balanceOf(IReserveToken reserveToken, address account) internal view returns (uint256) {
+        if (isNativeToken(reserveToken)) {
+            return account.balance;
+        }
+
+        return toIERC20(reserveToken).balanceOf(account);
+    }
+
+    /**
+     * @dev transfers a specific amount of the reserve token
+     *
+     * @param reserveToken the address of the reserve token
+     * @param to the destination address to transfer the amount to
+     * @param amount the amount to transfer
+     */
+    function safeTransfer(
+        IReserveToken reserveToken,
+        address to,
+        uint256 amount
+    ) internal {
+        if (amount == 0) {
+            return;
+        }
+
+        if (isNativeToken(reserveToken)) {
+            payable(to).transfer(amount);
+        } else {
+            toIERC20(reserveToken).safeTransfer(to, amount);
+        }
+    }
+
+    /**
+     * @dev transfers the whole balance of the reserve token
+     *
+     * @param reserveToken the address of the reserve token
+     * @param to the destination address to transfer the amount to
+     */
+    function safeTransfer(IReserveToken reserveToken, address to) internal {
+        safeTransfer(reserveToken, to, balanceOf(reserveToken, address(this)));
+    }
+
+    /**
+     * @dev transfers a specific amount of the reserve token
+     * this function ignores a reserve token which represents an ETH reserve
+     *
+     * @param reserveToken the address of the reserve token
+     * @param from the source address to transfer the amount from
+     * @param to the destination address to transfer the amount to
+     * @param amount the amount to transfer
+     */
+    function safeTransferFrom(
+        IReserveToken reserveToken,
+        address from,
+        address to,
+        uint256 amount
+    ) internal {
+        if (amount == 0 || isNativeToken(reserveToken)) {
+            return;
+        }
+
+        toIERC20(reserveToken).safeTransferFrom(from, to, amount);
+    }
+
+    /**
+     * @dev ensures that the spender has sufficient allowance
+     * this function ignores a reserve token which represents an ETH reserve
+     *
+     * @param reserveToken the address of the reserve token
+     * @param spender the address allowed to spend
+     * @param amount the allowed amount to spend
+     */
+    function ensureApprove(
+        IReserveToken reserveToken,
+        address spender,
+        uint256 amount
+    ) internal {
+        if (isNativeToken(reserveToken)) {
+            return;
+        }
+
+        toIERC20(reserveToken).ensureApprove(spender, amount);
+    }
+
+    /**
+     * @dev utility function that converts an IReserveToken to an IERC20
+     *
+     * @param reserveToken the address of the reserve token
+     *
+     * @return an IERC20
+     */
+    function toIERC20(IReserveToken reserveToken) private pure returns (IERC20) {
+        return IERC20(address(reserveToken));
```

Comment:
Maybe `IReserveToken` inheriting from `IERC20` (assuming that interface inheritance is already supported) would relieve us from this "double casting" (hence from this helper function altogether).

Reply:
I tried to emphasize this during our call today: `IReserveToken` is intentionally no inheriting from `IERC20` so that it would be harder to mix them (e.g., accidentally implicitly casting `IReserveToken` to `IERC20`, thus calling the wrong `balanceOf`, etc.).